src/Accounting/index.ts

```diff
@@ -0,0 +1,202 @@
+    quote,
+    base,
+    price,
+    maxLeverage,
+) => {
+    return totalMargin(base, quote, price) - (base !== 0 ? LIQUIDATION_GAS_COST * RYAN_6 + calcNotionalValue(base, price) / maxLeverage : 0);
+};
+
+/**
+ * Calculates the notional value of the position
+ * @param base Amount of base asset, this is considered the position of the account
+ * @param price The given price of the asset 
+ * @returns the notional value of the position
+ */
+export const calcNotionalValue: (base: number, price: number) => number = (base, price) => {
+    return Math.abs(base) * price;
+};
+
+/**
+ * Calculates the minimum margin required for a given position and price. This is closely related to the liquidationPrice
+ * @param quote Amount of quote asset
+ * @param base Amount of base asset, this is considered the position of the account
+ * @param price The given price of the asset 
+ * @returns the minimum margin required for an accounts outstanding position. 
+ *  An account with minimumMargin are in a position close to liquidation.
+ */
+export const calcMinimumMargin: (quote: number, base: number, price: number, maxLeverage: number) => number = (
+    quote,
+    base,
+    price,
+    maxLeverage,
+) => {
+    const margin = totalMargin(base, quote, price);
+    if (margin > 0 || base < 0) {
+        return LIQUIDATION_GAS_COST * RYAN_6 + calcNotionalValue(base, price) / maxLeverage;
+    } else {
+        return 0;
+    }
+};
+
+/**
+ * Total margin of an account given a position
+ *  aka equity, the amount owned by the user if they cashed out at the given price
+ * @param quote Amount of quote asset
+ * @param base Amount of base asset, this is considered the position of the account
+ * @param price The given price of the asset 
+ * @returns the total margin of an account
+ */
+export const totalMargin: (base: number, quote: number, price: number) => number = (base, quote, price) =>
+    quote + base * price;
+
+/**
+ * Calculates a theoretical market exposure if it took all the 'best' orders it could
+ *  Returns this exposure and the orders that allow it to gain this exposure
+ * @param quote Amount of quote asset
+ * @param leverage leverage of the trade this could be passed in as quote leverage * quote
+ */
+export const calcTradeExposure: (
+    quote: number,
+    leverage: number,
+    orders: FlatOrder[],
+) => { exposure: number, slippage: number, tradePrice: number } = (quote, leverage, orders) => {
+    if (orders.length) {
+        // weighted average of the price, where the weights are the amounts at each price
+        let exposure = 0,
+            sumOfWeights = 0,
+            totalUnits = 0;
+        let deposit = quote * leverage; // total units of underlying
+        for (const order of orders) {
+            const amount = order.amount;
+            const orderPrice = order.price;
+            // remainding units of accounts quote use
+            const r = deposit - amount * orderPrice;
```

Comment:
so, r is (total buying power - notional value). We need to include mainteannce margin somehow, too